code-collab-app/src/editorpage.js

```diff
@@ -32,70 +30,68 @@ class Editorpage extends React.Component {
-    });
+    this.props.peer_data.add_new_cell(cellContents);
+    // TODO - maybe call peer_data method to update other peers
   }
 
   // Locks a cell so the user cannot edit it
   // key: the index of the cell to lock
   lockEditorCell = (key) => {
-    let temp = this.state.cellLocked;
-    temp[key] = true;
-    this.setState({ cellLocked:temp });
+    // Lock cell so user may not edit
+    this.props.peer_data.set_cell_locked(key, true);
   }
 
-  // Unlocks a cell so the user may edit it
+  // Unlock a cell so the user may edit it
   // key: the index of the cell to unlock
   unlockEditorCell = (key) => {
-    let temp = this.state.cellLocked;
-    temp[key] = false;
-    this.setState({ cellLocked:temp });
+    this.props.peer_data.set_cell_locked(key, false);
+
   }
 
   // Requests lock on cell from all peers in session
   // key: index of cell
   requestEditorCellLock = (key) => {
     // TODO - implement socket.io request to lock this cell on all other peers in session
+    // Will probably involve calling lockEditorCell()
     console.log('Requesting cell lock from all peers for cell ' + key);
     return true;
   }
 
   // Broadcast cell update and release lock for all other peers in session
   // key: index of cell
   // cellContents: text contents of the cell update
-  broadcastCellUpdate = (key, cellContents) => {
+  broadcastCellUpdate = (key) => {
     // TODO - implement socket.io broadcast to update cell for all other peers and release lock
+    // Will probably involve calling unlockEditorCell()
     console.log('Broadcasting cell update to peers for cell ' + key);
   }
 
-  // Event handler for cursor entering a cell. Requests a lock on the cell from
-  // all peers
+  // Event handler for cursor entering a cell. Requests a lock on the cell from all peers.
   // e: onFocus event
   // key: index of cell
   selectCellEvent = (e, key) => {
     e.preventDefault();
-    // BROADCAST UPDATE AND RELEASE LOCK FROM LAST CELL EDITED
-    // Get text of cell just released from editing
-    const lastCellEdited = this.state.cellCurrentlyEditing;
-    if (lastCellEdited !== null) {
-      // Broadcast updates to cell just released from editing
-      let text = this.state.cellText[lastCellEdited]; // Select that cell and extract the text
-      this.broadcastCellUpdate(lastCellEdited, text);
-    }
-
-    // REQUEST LOCK ON NEW CELL
-    this.setState({ cellCurrentlyEditing:null });
+    // Request lock on cell from all peers
     const lockApproved = this.requestEditorCellLock(key);
     if (lockApproved === false) return;
-    // Save the cell currently being edited
-    this.setState({ cellCurrentlyEditing:key });
+    // Remember the cell currently being edited
+    this.props.peer_data.set_current_cell(key);
+  }
+
+  // Event handler for cursor leaving a cell. Broadcasts update for that cell and releases lock.
+  // e: onBlur event
+  // key: index of cell
+  leaveCellEvent = (e, key) => {
```

Comment:
key is an unused arg here and can be removed

Reply:
Okay, I will remove it.